### ISSUE TITLE
Gaussian differential privacy and its conversion to (epsilon, delta)-DP

### ISSUE THREAD

**Issue by lxuechen** [pytorch/opacus#187]
Gaussian differential privacy is starting to be adopted by researchers when accounting loss in practice. It would be nice if the library incorporated utility functions of computing this, even though tf_privacy already has code for this, and the code for accounting loss for SGM is not hard to write.

**Comment by ffuuugor** [comment]
@lxuechen do you refer to the same calculations as in #186 ?

**Comment by lxuechen** [comment]
No, these are distinct procedures. Gaussian differential privacy is different from RDP, and converting GDP to (eps, del)-DP typically results in small epsilons when a type of CLT holds (when each step doesn't incur too much loss in privacy, and the ultimate algorithm is a composition of many updates). 

See [this](https://arxiv.org/abs/1911.11607) for reference, and [this](https://github.com/tensorflow/privacy/blob/master/tensorflow_privacy/privacy/analysis/gdp_accountant.py) for its implementation in the tf_privacy codebase.

**Comment by gkaissis** [comment]
We have GDP and accounting implemented in our DP-SGD library, feel free to check out the code [here](https://github.com/gkaissis/deepee).

There can be some details which make GDP a bit difficult to use for users. For example, the optimization procedure to convert the mu parameter of GDP to epsilon/delta can sometimes fail when the users input "unrealistic" values for the privacy parameters to try to "disable" DP, which RDP doesn’t suffer from. We have a fallback to RDP implemented (actually using the opacus accounting) for that case. It may therefore be a bit of a fuss sometimes which could deter users with less experience. Very happy to answer questions!
